lib/constructs/cohere-embded-v3-app-runner.ts

```diff
@@ -107,8 +107,8 @@ export class CdkCohereEmbedV3AppRunnerStack extends cdk.NestedStack {
         // apply removal policy to appRunnerTaskRole
         appRunnerTaskRole.applyRemovalPolicy(cdk.RemovalPolicy.DESTROY);
 
-        const cpu = parsingCpuType(props.cdkDeployCpuType);
-        const memory = parsingMemoryType(props.cdkDeployMemoryType);
+        const cpu = parseCpuType(props.cdkDeployCpuType);
+        const memory = parseMemoryType(props.cdkDeployMemoryType);
```

Comment:
**Suggestion:** Consider adding a validation check for `props.cdkDeployCpuType` and `props.cdkDeployMemoryType` before passing them to `parseCpuType` and `parseMemoryType` respectively. This will help catch any invalid values early and provide a more informative error message.
> 🚀  **High Priority** 
 > [validation, importance: 8]
```suggestion
        const cpu = parseCpuType(props.cdkDeployCpuType);
        if (!cpu) {
            throw new Error(`Invalid CPU type: ${props.cdkDeployCpuType}`);
        }
        const memory = parseMemoryType(props.cdkDeployMemoryType);
        if (!memory) {
            throw new Error(`Invalid Memory type: ${props.cdkDeployMemoryType}`);
        }
```